include/podio/CollectionBuffers.h

```diff
@@ -41,7 +42,9 @@ struct CollectionWriteBuffers {
 };
 
 struct CollectionReadBuffers {
+  bool needsSchemaEvolution{false};
   void* data{nullptr};
+  void* data_oldschema{nullptr};
```

Comment:
It depends on how we want to split up the transformation in the long run.
A possibility is to give it an additional state whether a transformation is still needed and do it in place

Reply:
But doing it in place would mean to manipulate the `data` buffer, right? 

In principle in the way I imagine the actual schema evolution, you send in a `data` buffer into the black box and you get one back. Inside this black box we are free to do whatever we want, even simply nothing. However, i would like to avoid the situation where we have data in two different versions hanging around, as that is an easy way towards somewhat undefined state(s).

Reply:
Yes, the black box approach is what I wanted. At the moment (till re-factoring) we however do not have good entities to book keep the data buffers before and after though. 
To me an item that goes into the todos